app/assets/javascripts/discourse/app/components/time-shortcut-picker.js

```diff
@@ -134,7 +133,7 @@ export default Component.extend({
     if (lastTime && lastDate) {
       let parsed = parseCustomDatetime(lastDate, lastTime, this.userTimezone);
 
-      if (parsed < now(this.userTimezone)) {
+      if (parsed < now(this.userTimezone) || !parsed.isValid()) {
```

Comment:
What do you think about changing order here? First ensure that new parsed value is valid and then compare it with current time